microceph/common/network.go

```diff
@@ -0,0 +1,73 @@
+package common
+
+import (
+	"fmt"
+	"net"
+)
+
+// FindIpOnSubnet scans the host's network interfaces to check if an IP is available
+// for the provided subnet. It returns the found IP address or an empty string.
+func FindIpOnSubnet(subnet string) (string, error) {
+	_, sn, err := net.ParseCIDR(subnet)
+	if err != nil {
+		return "", err
+	}
+
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return "", err
+	}
+
+	for _, iface := range ifaces {
+		addrs, _ := iface.Addrs()
+		for _, addr := range addrs {
+			ipNet, ok := addr.(*net.IPNet)
+			if !ok {
+				continue
+			}
+
+			if !ipNet.IP.IsGlobalUnicast() {
+				continue
+			}
+
+			if sn.Contains(ipNet.IP) {
+				return ipNet.IP.String(), nil
+			}
+		}
+	}
+	return "", fmt.Errorf("no IP belongs to provided subnet %s", subnet)
+}
+
+// FindNetworkAddress locates the provided IP address on host's network interfaces.
+// It returns the containing subnet address on success or an empty string on failure.
+func FindNetworkAddress(address string) (string, error) {
+	// Parse provided address.
+	monIp := net.ParseIP(address)
```

Comment:
If `address` is invalid we'd be getting back a nil monIp, suggest to check for that to avoid a nil panic

Reply:
good catch, thanks 😄